inc/az_ulib_ustream_forward_base.h

```diff
@@ -0,0 +1,339 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+#ifndef AZ_ULIB_USTREAM_FORWARD_BASE_H
+#define AZ_ULIB_USTREAM_FORWARD_BASE_H
+
+#include "az_ulib_base.h"
+#include "az_ulib_config.h"
+#include "az_ulib_pal_os.h"
+#include "az_ulib_result.h"
+#include "azure/az_core.h"
+
+#ifdef __cplusplus
+#include <cstddef>
+#include <cstdint>
+#else
+#include <stddef.h>
+#include <stdint.h>
+#endif /* __cplusplus */
+
+#include "azure/core/_az_cfg_prefix.h"
+
+#ifndef AZ_ULIB_USTREAM_BASE_H
+/**
+ * @brief   Define offset_t with the same size as size_t.
+ */
+typedef size_t offset_t;
+
+/**
+ * @brief   Signature of the function to release memory passed to the ustream_forward
+ *
+ * @param[in]   release_pointer       void pointer to memory that needs to be free'd
+ *
+ */
+typedef void (*az_ulib_release_callback)(void* release_pointer);
+#endif /* AZ_ULIB_USTREAM_BASE_H */
+
+/**
+ * @brief   Forward declaration of az_ulib_ustream_forward. See #az_ulib_ustream_forward_tag for
+ *          struct members.
+ */
+typedef struct az_ulib_ustream_forward_tag az_ulib_ustream_forward;
+
+/**
+ * @brief Signature of the function to be invoked by the `ustream_forward->_internal.flush` operation when  
```

Comment:
>ustream_forward->_internal.flush [](http://example.com/codeflow?start=58&length=32)

Please do not document access to internal members of a structure. Replace it by the external API, in this case `az_ulib_ustream_forward_flush`.